# ü¶† Bioinformatics Algorithm 
## An  understanding of how Biopython and its algorithm works generally on Python
I am sharing my code repository to understand the basic working and principle of various BioInformatics code in Python; I am writing this github repository with 2 main intensions of mine first and foremost is to build myself a code base to look back onto how I learnt the Bioinformatics and understood how the skeleton of code works/interacts with my subject of interest i.e. Biology and second is to make you people understand how the BioPython a popular Biological library in Python which almost feel like a cheat sheet to solve complex algorithms in Biologial genetic data.

Hence, Let's dive into the journey with me where I start from basic to the complex concepts which is actually necessary to solve the real world calculations in Bioinformatics.

## üß¨ [Season 1 - Diggin down deep](Season_1)
A very basic initial Structural concept explained which primarirly involves this concepts:
1. Calculate pattern
2. Frequecney map of DNA sequence
3. Most frequent K-mers
4. Complementary DNA string
5. Pattern Matching

This all seems very intimaidating at first but trust me all of 'em have been explained in a very interesting manner, I urge you to take a look at it and you'll surely thank me later considering that.

## üß¨ [Season 2 - Adding steel rods as skeleton for the building](Season_2)
A bit more complex and tough concepts are being touched upon which needed a lot more efforts to be understood are taken into account here, and I am giving my level best to make it most simpler interpretation of that concept to understand and trying to address each and every question which I had and possibly you could have during your learning journey.

The primary topics involved here looks like:
1. Symbol Array in DNA Seq
2. Symbol Array but faster (Extended ``SymbolArray``)
3. Skew Array in DNA seq (2 methods explained)
4. Minimum Skew in DNA seq
5. Calculating Hamming Distance between DNA
6. Approximate Pattern Matching in DNA

## üß¨ [Season 3 - Adding Concrete slurry in the Skeletal Structure](Season_3)
Now season 3 and the upcoming last season involves a bit of Umbrella type of Code where I will walk you through each and every type of functions separately and then finally after using that small function codes we will make something very meaningful out of it. 

Therefore, I encourage you to just stick with me and just enjoy the miracle or magic that you are about to experience at the end during this journey.

The primary topics covered here are:
1. Basic Numpy Array understanding
2. Counting Nucleotide Frequencies in DNA
3. Profile Matrix from DNA
4. Consensus Sequence from DNA
5. Scoring DNA Motifs
6. Profile Most Probable K-mer
7. Greedy Motif Search (the finale of all - the main standalone function which uses all the above function in the form of Greedy Algorithm)
8. Finding Patterns with Mismatches (extras)

## üß¨ [Season 4 - Colouring the Building and Final touch-ups](Season_4)
Now the Series finale is this particular season which leads to end of this particualar git repo and in turn ends your initial journey of understanding very major Bioinformatics algorithms as well.

Here primarirly we will cover 2 important topics 
1. GreedyMotifSearch with Pseudocounts
2. Randomized Motif Search


### üöÄ And with that, this repository comes to a wrap!  
But wait, there's more! Keep an eye out for other exciting repositories on **Bioinformatics**, **Computational Drug Design**, and **Next-Gen Sequencing** over on my GitHub.  

That's all from my side for now. Until next time, this is your man **Mohit @mhtjsh** signing off. Cheers! ‚úåÔ∏è  
